src/platform/plugins/shared/share/public/components/export_popover.tsx

```diff
@@ -0,0 +1,313 @@
+        button={anchorElement!}
+        closePopover={onClose}
+        panelPaddingSize="none"
+      >
+        <EuiListGroup>
+          {shareMenuItems.map((menuItem) => (
+            <EuiToolTip
+              position="left"
+              content={selectedMenuItem?.config.toolTipContent}
+              key={menuItem.id}
+            >
+              <EuiListGroupItem
+                iconType={menuItem.config.icon}
+                key={menuItem.id}
+                label={menuItem.config.label}
+                data-test-subj={`exportMenuItem-${menuItem.config.label}`}
+                isDisabled={menuItem.config.disabled}
+                onClick={() => {
+                  setSelectedMenuItemId(menuItem.id);
+                  setIsFlyoutVisible(true);
+                }}
+              />
+            </EuiToolTip>
+          ))}
+        </EuiListGroup>
+      </EuiWrappingPopover>
+      {isFlyoutVisible && (
+        <EuiFlyout
+          data-test-subj="exportItemDetailsFlyout"
+          size="s"
+          onClose={flyoutOnCloseHandler}
+          css={() => ({
+            ['--euiFixedHeadersOffset']: 0,
+          })}
+          ownFocus
+          maskProps={{
+            headerZindexLocation: 'above',
+          }}
+        >
+          <EuiFlyoutHeader hasBorder>
+            <EuiTitle>
+              <h2>
+                <FormattedMessage
+                  id="share.export.flyoutTitle"
+                  defaultMessage="Export {objectType} as {type}"
+                  values={{ objectType, type: selectedMenuItem?.config.label }}
+                />
+              </h2>
+            </EuiTitle>
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody>
+            <EuiFlexGroup direction="column">
+              <Fragment>
+                {selectedMenuItem?.config.renderLayoutOptionSwitch && (
+                  <EuiFlexItem>
+                    <LayoutOptionsSwitch
+                      usePrintLayout={usePrintLayout}
+                      printLayoutChange={(evt) => setPrintLayout(evt.target.checked)}
+                    />
+                  </EuiFlexItem>
+                )}
+              </Fragment>
+              <Fragment>
+                {selectedMenuItem?.config.renderCopyURIButton && publicAPIEnabled && (
+                  <EuiFlexItem>
+                    <EuiFormRow
+                      label={
+                        <EuiText size="s">
+                          <h4>
+                            <FormattedMessage
+                              id="share.export.postURLHeading"
+                              defaultMessage="Post URL"
+                            />
+                          </h4>
+                        </EuiText>
+                      }
+                      fullWidth
+                    >
+                      <EuiFlexGroup direction="column">
+                        <EuiFlexItem>
+                          <EuiText size="s" color="subdued">
```

Comment:
I wonder for Dashboard JSON export if we could provide our own Component instead of injecting our content into a code block designed for URLs?

Maybe we can add an optional `generateAssetComponent` function on the `ExportShare['config']`? Then it would make my other comments below unnecessary.

Is that out of scope for this PR?

Reply:
You're right... the code block makes the assumption that the content would be very much similar to the content we've had in the past, I think for sake of upholding visual consistency across export types I'd rather we didn't go the route of passing a component, instead we could opt for more customization points. I'm thinking `title`, `helpText` if needed and then `language` value for the code editor.

Reply:
This change has been made in https://github.com/elastic/kibana/pull/217109/commits/fee13662c682aeca242cb1ac4e7adf096ec46990

Reply:
@nickpeihl looks like we'd get that `generateAssetsComponent` prop, I spoke with response ops folks, and their use case for export can't be handled without providing a custom outlet.

Reply:
This is great! Thank you!